jconstraints-core/src/main/java/gov/nasa/jpf/constraints/expressions/PropositionalCompound.java

```diff
@@ -58,10 +59,19 @@ public Boolean evaluate(Valuation values) {
 
   @Override
   public Boolean evaluateSMT(Valuation values) {
-    boolean lv = left.evaluateSMT(values);
-    boolean rv = right.evaluateSMT(values);
+    boolean lv = false, lvUndecided = false, rv = false, rvUndecided = false;
+    try {
+      lv = left.evaluateSMT(values);
+    } catch (UndecidedBooleanExeception e) {
+      lvUndecided = true;
+    }
+    try {
+      rv = right.evaluateSMT(values);
+    } catch (UndecidedBooleanExeception e) {
+      rvUndecided = true;
+    }
 
-    return operator.eval(lv, rv);
+    return rvUndecided || lvUndecided ? true : operator.eval(lv, rv);
```

Comment:
Simplification possible (via IDE warning)
```suggestion
    return rvUndecided || lvUndecided || operator.eval(lv, rv);
```